menpo/image/base.py

```diff
@@ -2551,28 +2609,10 @@ def scale_func(pixels, axis=None):
         return self._normalize(scale_func, mode=mode)
 
     def _normalize(self, scale_func, mode='all'):
-        new = self.copy()
-        new._normalize_inplace(scale_func, mode=mode)
-        return new
-
-    def _normalize_inplace(self, scale_func, mode='all'):
-        pixels = self.as_vector(keep_channels=True)
-        if mode == 'all':
-            centered_pixels = pixels - np.mean(pixels)
-            scale_factor = scale_func(centered_pixels)
-
-        elif mode == 'per_channel':
-            centered_pixels = pixels - np.mean(pixels, axis=1)[..., None]
-            scale_factor = scale_func(centered_pixels, axis=1)[..., None]
-        else:
-            raise ValueError("mode has to be 'all' or 'per_channel' - '{}' "
-                             "was provided instead".format(mode))
-
-        if np.any(scale_factor == 0):
-            raise ValueError("Image has 0 variance - can't be "
-                             "normalized")
-        else:
-            self._from_vector_inplace(centered_pixels / scale_factor)
+        from menpo.feature import normalize
```

Comment:
To be clear:

- `normalise_pixels_range`
- `normalize_std`
- `normalize_norm`

Also, are we going to continue to support these as methods on Image in addition to their role as features? I think I'm right in saying we removed `Image.gradient()` in favour of `gradient(img)`. Don't have too strong feelings on this, other than that we should be consistent.